shell/platform/linux/fl_renderer_gl.cc

```diff
@@ -0,0 +1,122 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_renderer_gl.h"
+
+#include "flutter/shell/platform/linux/fl_backing_store_provider.h"
+#include "flutter/shell/platform/linux/fl_view_private.h"
+
+struct _FlRendererGL {
+  FlRenderer parent_instance;
+};
+
+G_DEFINE_TYPE(FlRendererGL, fl_renderer_gl, fl_renderer_get_type())
+
+// Implements FlRenderer::create_contexts.
+static gboolean fl_renderer_gl_create_contexts(FlRenderer* renderer,
+                                               GtkWidget* widget,
+                                               GdkGLContext** visible,
+                                               GdkGLContext** resource,
+                                               GError** error) {
+  GdkWindow* window = gtk_widget_get_parent_window(widget);
+
+  *visible = gdk_window_create_gl_context(window, error);
+  *resource = gdk_window_create_gl_context(window, error);
```

Comment:
Looks sus to use error twice without checking it. probably best to check `error` after each call.

In practice I don't think this is unsafe or a memory leak, but if the 1st call results in an error the 2nd call will fail a non-fatal assertion inside GTK (which prints an error message to console). Might leak an error if assertions are turned off.

Reply:
Added error check right after line 24.